core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt

```diff
@@ -0,0 +1,429 @@
+import com.ustadmobile.core.account.LearningSpace
+import com.ustadmobile.core.domain.blob.savepicture.EnqueueSavePictureUseCase
+import com.ustadmobile.core.domain.localaccount.GetLocalAccountsSupportedUseCase
+import com.ustadmobile.core.domain.passkey.CreatePasskeyParams
+import com.ustadmobile.core.domain.passkey.CreatePasskeyUseCase
+import com.ustadmobile.core.domain.person.AddNewPersonUseCase
+import com.ustadmobile.core.impl.UstadMobileSystemCommon
+import com.ustadmobile.core.impl.appstate.AppUiState
+import com.ustadmobile.core.impl.appstate.LoadingUiState
+import com.ustadmobile.core.impl.appstate.Snack
+import com.ustadmobile.core.impl.config.GenderConfig
+import com.ustadmobile.core.impl.config.SystemUrlConfig
+import com.ustadmobile.core.impl.locale.entityconstants.PersonConstants
+import com.ustadmobile.core.impl.nav.UstadSavedStateHandle
+import com.ustadmobile.core.util.MessageIdOption2
+import com.ustadmobile.core.util.ext.appendSelectedAccount
+import com.ustadmobile.core.view.SiteTermsDetailView
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.core.viewmodel.UstadEditViewModel
+import com.ustadmobile.core.viewmodel.clazz.list.ClazzListViewModel
+import com.ustadmobile.core.viewmodel.person.child.AddChildProfileViewModel
+import com.ustadmobile.core.viewmodel.person.edit.PersonEditViewModel
+import com.ustadmobile.core.viewmodel.person.edit.PersonEditViewModel.Companion.ARG_REGISTRATION_MODE
+import com.ustadmobile.door.ext.doorIdentityHashCode
+import com.ustadmobile.door.ext.doorPrimaryKeyManager
+import com.ustadmobile.door.util.systemTimeInMillis
+import com.ustadmobile.lib.db.entities.Person
+import com.ustadmobile.lib.db.entities.Person.Companion.GENDER_UNSET
+import com.ustadmobile.lib.db.entities.PersonPicture
+import com.ustadmobile.lib.db.entities.ext.shallowCopy
+import io.github.aakira.napier.Napier
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.kodein.di.DI
+import org.kodein.di.direct
+import org.kodein.di.instance
+import org.kodein.di.instanceOrNull
+import org.kodein.di.on
+
+
+data class SignUpUiState(
+
+    val person: Person? = null,
+
+    val password: String? = null,
+
+    val genderOptions: List<MessageIdOption2> = PersonConstants.GENDER_MESSAGE_IDS_AND_UNSET,
+
+    val personPicture: PersonPicture? = null,
+
+    val registrationMode: Int = 0,
+
+    val usernameError: String? = null,
+
+    val fullName: String? = null,
+
+    val passwordError: String? = null,
+
+    val dateOfBirthError: String? = null,
+
+    val parentContactError: String? = null,
+
+    val genderError: String? = null,
+
+    val firstNameError: String? = null,
+
+    val lastNameError: String? = null,
+
+    val isParent: Boolean = false,
+
+    val isTeacher: Boolean = false,
+
+    val signupWithPasskey: Boolean = true,
```

Comment:
This name description needs to be clarified. signup with passkey is a verb to do something, nota state. Maybe passkeySupported or signupWithPasskeyVisible ?